packages/core/src/StateMachine.ts

```diff
@@ -377,16 +391,43 @@ export class StateMachine<
       | State<TContext, TEvent, TResolvedTypesMeta>
       | StateConfig<TContext, TEvent>
   ): State<TContext, TEvent, TResolvedTypesMeta> {
+    const configuration =
+      stateConfig.configuration ??
+      getConfiguration(getStateNodes(this.root, stateConfig.value));
+
     const state =
       stateConfig instanceof State
         ? stateConfig
-        : (new State(stateConfig) as State<
-            TContext,
-            TEvent,
-            TResolvedTypesMeta
-          >);
+        : new State(
+            { ...stateConfig, configuration: Array.from(configuration) },
```

Comment:
```suggestion
            { ...stateConfig, configuration },
```
Would this work? Not sure why we need to use `Array.from` here?

Reply:
This one is coming from the `getConfiguration` call a couple of lines above - and that one returns a set, so it has to be converted to an array here.

OTOH... according to TS, this `getConfiguration` won't be called as `stateConfig.configuration` is always there. It makes this `Array.from` redundant as it's actually always an array (at least according to TS).

Reply:
The types here were wrong (`stateConfig.configuration` may not always exist; the types lied! Sorry)

Fixed here: 110c8096be1eaea1314030f897da010f6e23a5ff